controller/controller.go

```diff
@@ -45,11 +66,13 @@ type Controller struct {
 func (c *Controller) RunOnce() error {
 	records, err := c.Registry.Records()
 	if err != nil {
+		registryErrors.Inc()
 		return err
 	}
 
 	endpoints, err := c.Source.Endpoints()
 	if err != nil {
+		sourceErrors.Inc()
```

Comment:
`c.Registry.ApplyChanges` below can also return an `err`. Do you want to count them as well?

These two are `reading` from Kubernetes and the DNS provider, whereas `ApplyChanges` is also writing to the DNS provider.

Reply:
So I wasn't feeling confident to add metrics to all DNS providers. Instead I've added a dimension to total errors clearly indicating these are read errors.

Reply:
You don't have to do it for each provider. Just catching the error [here](https://github.com/kubernetes-incubator/external-dns/blob/v0.5.4/controller/controller.go#L64) like you already do above should be fine.

Reply:
Ah right, done so